src/components/ChatWindow/MessageList/MessageList.tsx

```diff
@@ -28,11 +28,19 @@ export default function MessageList({ messages }: MessageListProps) {
         const time = message.dateCreated
           .toLocaleTimeString('en-us', { hour: 'numeric', minute: 'numeric' })
           .toLowerCase();
+
+        const previousTime = messages[idx - 1]?.dateCreated
+          .toLocaleTimeString('en-us', { hour: 'numeric', minute: 'numeric' })
+          .toLowerCase();
```

Comment:
quick question, since `time` and `previousTime` are almost identical, would this be better as a function that accepts a time and we could pass in `message.dateCreated` or `messages[idx-1]?.dateCreated`? or maybe that would just make it more confusing 🤔

Reply:
Good question!

There is some repeated code here. In my opinion, I think it can be fine to have some duplicate code. When code is repeated just once, I sometimes feel that it isn't worth it to create an abstraction in order to get rid of the duplicated code. Sometimes you end up with more code than you started with. 

However, I tried it out in this instance, and with your idea, I found that I was able to both reduce the amount of code in this file and make it more readable at the same time. So here it definitely helps! I'll upload my change.